### ISSUE TITLE
Fedora 24 yum packages

### ISSUE THREAD

**Issue by mackeyja92** [docker/docker#23259]
Fedora 24 is not on the `https://yum.dockerproject.org/repo/main/fedora/` repos. Subsequently trying to install docker-engine with yum is failing. Fedora 24 is currently in public beta and will be going for GA later this month.

**Comment by runcom** [comment]
I think we're waiting on GA anyway (since Docker always did this in the past iinm)

**Comment by mackeyja92** [comment]
@runcom Ok, what is the recommended way to proceed in the meantime?

**Comment by runcom** [comment]
Either build the rpm yourself (we have a make target iirc) or compile and install from source - or probably grab older or experimental Linux builds. 

I'll defer this to @thaJeztah you may know more than me

**Comment by thaJeztah** [comment]
Generally, we try to have support for distro releases that are released the moment we do our release. I know we once ran into an issue with Fedora 23, which was released literally 14 hours after we did a Docker release, so as an exception, we published packages for that after we did the release.

The same accounts for EOL versions; if we know a version is EOL'd, we no longer publish packages for that version, because users will no longer receive security updates from upstream, so it's no longer suitable to run those versions in production.

I must add here, that we rely on official images for a distro version to be available on Docker Hub, because we build the packages in a container, using the official images

According to https://fedorapeople.org/groups/schedule/f-24/f-24-key-tasks.html and https://fedoraproject.org/wiki/Releases/24/Schedule, "Final Release Public Availability (GA)" for Fedora 24 is scheduled for Tuesday, June 14th.

Also, Fedora 22 EOL date looks to be Tuesday, July 14th, so we should have an eye on that, and decide if we're going to be releasing Docker 1.12 for that (depending on the release date for Docker 1.12 - I don't have the exact date yet).

So, yes, I think we should aim for having packages for Fedora 24 available for Docker 1.12, *however*, there's no official Image yet on Docker Hub https://hub.docker.com/r/library/fedora/tags/

After an official image is available, a Dockerfile can be added (similar to https://github.com/docker/docker/blob/master/contrib/builder/rpm/amd64/fedora-23/Dockerfile), so that packages are built.

I'll add this to the 1.12 milestone

@tianon perhaps you know when work is started for the official Fedora 24 images?

**Comment by thaJeztah** [comment]
Also note that this is all for the OSS / Community edition of Docker. The commercially supported ('CS') version of Docker may keep different schedules for supported distros, and is known in some cases to include support for distros that are officially EOL'd

**Comment by mackeyja92** [comment]
@thaJeztah Ok I will be looking forward to those releases. I tried to build it myself last night but was unsuccessful. Is there a guide somewhere for building docker without docker? I need to get this running soon. Thanks

**Comment by thaJeztah** [comment]
@mackeyja92 you may be able to find some info in https://github.com/docker/docker/blob/master/project/PACKAGERS.md, but it could be outdated due to recent changes (multiple binaries, RunC and containerd integration). The best place to look may be the Dockerfile and see what steps are used to build those; https://github.com/docker/docker/blob/master/Dockerfile

**Comment by den-is** [comment]
since [ fedora24](https://torrents.fedoraproject.org/) is already released hope to see Docker repo updated as soon as possible.

IMO: why not to prepare repo for RC or even Beta fedora releases?

**Comment by piotrantosik** [comment]
fedora 24 is relased, package is available https://hub.docker.com/r/library/fedora/tags/, @thaJeztah it is possible release docker 1.11 for fedora 24?

**Comment by doenietzomoeilijk** [comment]
Is it possible that this somehow regressed? I'm on FC24 and I was able to install Docker through the instructions provided (using the provided `docker.repo` file, but after a while I started getting the _Failed to synchronize cache for repo_ messages from dnf. Manually checking the URL tells me it goes up to 23, not 24, so it looks like something went missing on `yum.dockerproject.org`.

Any ideas why? Did I miss anything?

**Comment by thaJeztah** [comment]
@doenietzomoeilijk docker 1.12 is the first release for Fedora 24, and hasn't been released yet; you can try the release candidates (test.docker.com), or try manually installing the regular release, but picking the Fedora 23 repo

**Comment by rlpowell** [comment]
No, I'm *quite* certain that @doenietzomoeilijk is correct: the instructions at https://docs.docker.com/engine/installation/linux/fedora/ worked fine a couple of weeks ago (I used them repeatedly), and now they don't.  Using the repo file there now gives:

Failed to synchronize cache for repo 'dockerrepo', disabling.                                 
Last metadata expiration check: 0:00:09 ago on Thu Jul 28 00:03:26 2016.                      

, and it takes a while to do so; this was not the case before.

So something definitely changed.

Regardless, is there some reason you can't have a 24/ directory with 1.11.2 in it?

**Comment by thaJeztah** [comment]
Nothing should have changed in the 1.11.2 repo; are you sure you were running Fedora 24 when it worked? Or did you update your machine?

> Regardless, is there some reason you can't have a 24/ directory with 1.11.2 in it?

Our current policy is that we only maintain a single release, and don't back port versions; however, docker 1.12.0 should be released tomorrow, and will be available for Fedora 24

**Comment by xasx** [comment]
@rlpowell 
> Regardless, is there some reason you can't have a 24/ directory with 1.11.2 in it?

The point here is that it did *not* work for me. I could not get docker to start after installing it in F24 from F23 repos. So developers would have to take additional effort to make it work…

**Comment by aaraodeo** [comment]
I changed to baseurl=https://yum.dockerproject.org/repo/main/fedora/24/ in docker.repo
then i am able to install docker-engine 1.12.1-1 in fc24 by using following command line:
dnf install docker-engine --allowerasing

**Comment by aaraodeo** [comment]
But to get integration of docker with cockpit i reverted back to default docker 1.10.3 using the following command line:
dnf install cockpit-docker --allowerasing

**Comment by xasx** [comment]
Well, `cockpit-docker` is not compatible with `docker-engine` from Docker's repos, but requires a `docker` package provided by the `updates` repo.

Also, there's some docker-migrator involved:
`dnf install cockpit-docker --allowerasing --assumeno`
```
================================================================================================================================================================
 Package                                    Arch                        Version                                          Repository                        Size
================================================================================================================================================================
Installing:
 cockpit-docker                             x86_64                      0.117-1.fc24                                     updates                          119 k
 docker                                     x86_64                      2:1.10.3-50.gita612434.fc24                      updates                          6.7 M
 docker-selinux                             x86_64                      2:1.10.3-50.gita612434.fc24                      updates                           75 k
 docker-v1.10-migrator                      x86_64                      2:1.10.3-50.gita612434.fc24                      updates                          1.9 M
 oci-register-machine                       x86_64                      0-1.1.git7d4ce65.fc24                            fedora                           925 k
 oci-systemd-hook                           x86_64                      0.1.4-1.fc24                                     fedora                            30 k
Removing:
 docker-engine                              x86_64                      1.12.1-1.fc24                                    @dockerrepo                       79 M
 docker-engine-selinux                      noarch                      1.12.1-1.fc24                                    @dockerrepo                       44 k
```

@aaraodeo Probably you want to file another separate issue regarding the proper cockpit integration.

**Comment by aaraodeo** [comment]
I completely agree with @xasx 
Eagerly waiting for docker and cockpit update in fedora 24 repo as new releases are available in respective  docker and cockpit repositories.
